packages/react/src/components/Modal/Modal.js

```diff
@@ -220,12 +220,12 @@ export default class Modal extends Component {
   }
 
   initialFocus = focusContainerElement => {
-    const primaryFocusElement = focusContainerElement
-      ? focusContainerElement.querySelector(this.props.selectorPrimaryFocus)
-      : null;
-    if (primaryFocusElement) {
-      return primaryFocusElement;
+    if (!focusContainerElement && this.props.focusTrap) {
```

Comment:
Do you want to assign `this.innerModal.current` to `focusContainerElement` if `focusContainerElement` isn't there, given missing `focusContainerElement` seems your concern? This function is called for `this.props.focusTrap === false` scenario, so we cannot re-write this function for `this.props.focusTrap === true` scenario. Thanks!

Reply:
Changed in https://github.com/carbon-design-system/carbon/pull/4172/commits/90c86911f1be36421b4b41b90cd478e26111d61e

Reply:
Would you want to keep NPE guard for `containerElement` just in case? It was originally for before-mount/after-unmount scenarios (though there may not be the case).

Reply:
Added the guard back in https://github.com/carbon-design-system/carbon/pull/4172/commits/742ebde8eccedeb31c8a2c44ec9fa7301417c993